opengovsg/askgovsg
# feat(image-upload): update image upload modal

## Problem

The current image upload tool is not intuitive.

Working towards #746 

## Solution

Create a custom image component for the rich text editor, referencing updated design of image modal in [Figma](https://www.figma.com/file/O6lxjjfn04Ow6nn2TMStpk/AskGov-Design-Master-v2?node-id=6271%3A69638)
Main changes include:
- Image preview, file name and size are shown for uploaded image
- Alt text input is displayed only if image source exists
- Spinner indicates loading image status
- Trash icon provided to remove image
- Image icon in RTE is only visible in desktop view
- URL upload option has been removed due to Content Security Policy (see comment below)

Additionally, Chakra theming is used to provide styles

## Before & After Screenshots

**BEFORE**:

https://user-images.githubusercontent.com/56983748/142226804-8c95e108-22f0-4695-b421-936575698541.mov



**AFTER**:


https://user-images.githubusercontent.com/56983748/142381616-915521bd-14b6-48a0-96ba-62cd5588fd71.mov






## Tests

- Insert image using file upload
- Modal should be cleared upon clicking "Cancel" or "Insert Image"
- Try to upload an image without keying in alt text - you should not be able to click the "Insert Image" button
- Inspect image, alt text should be whatever was entered
